src/swarmpal/toolboxes/tfa/tfa_processor.py

```diff
@@ -15,192 +15,662 @@
+    ):
+        # Select variable and set flag and maximum acceptable flag value
+        self.VARNAME = varname
+        if self.VARNAME == "B_NEC":
+            self.FLAGNAME = "Flags_B"
+            self.MAX_FLAG_VAL = 30
+        elif self.VARNAME == "F":
+            self.FLAGNAME = "Flags_F"
+            self.MAX_FLAG_VAL = 63
+        elif self.VARNAME == "E_NEC":
+            self.FLAGNAME = "Flags_TII"
+            self.MAX_FLAG_VAL = 23
+        elif self.VARNAME == "n":
+            self.FLAGNAME = "Flags_LP"
+            self.MAX_FLAG_VAL = 63
+        elif self.VARNAME == "Bubble_Probability":
+            self.FLAGNAME = "Flags_F"
+            self.MAX_FLAG_VAL = 1000
+        elif self.VARNAME == "FAC":
+            self.FLAGNAME = "Flags_F"
+            self.MAX_FLAG_VAL = 1000
+        elif self.VARNAME == "EEF":
+            self.FLAGNAME = "Flags"
+            self.MAX_FLAG_VAL = 1000
+        elif self.VARNAME == "Eh_XYZ":
+            self.FLAGNAME = "Quality_flags"
+            self.MAX_FLAG_VAL = 0
+        elif self.VARNAME == "Ev_XYZ":
+            self.FLAGNAME = "Quality_flags"
+            self.MAX_FLAG_VAL = 0
+        else:
+            sys.exit("TfaInput: unrecognized 'varname' input: " + varname)
+
+        self.DEFAULTS["measurements"] = [self.VARNAME, self.FLAGNAME]
+
+        # Check if CHAOS field removal has been requested and if yes, add
+        # it to the input parameters so that it will be retrieved
+        if remove_chaos:
+            self.REMOVE_CHAOS = True
+            self.DEFAULTS["model"] = "'CHAOS-Core' + 'CHAOS-Static'"
+        else:
+            self.REMOVE_CHAOS = False
+            self.DEFAULTS["model"] = "None"
+
+        # set pad_times to either 3 hours, if Low-res data), or 5 minutes if
+        # high-res
+        self.COLLECTION = kwargs["collection"]
+        if "HR" in self.COLLECTION:
+            self.DEFAULTS["pad_times"] = [
+                dt.timedelta(minutes=5),
+                dt.timedelta(minutes=5),
+            ]
+        else:
+            self.DEFAULTS["pad_times"] = [dt.timedelta(hours=3), dt.timedelta(hours=3)]
+
+        # Parse sampling_time argument to float and set the class variable
+        self.SAMPLING_TIME = isotime2num(sampling_time)
+        self.DEFAULTS["sampling_step"] = sampling_time
+
+        # set time limits (user-requeste) and pad for retrieval
+        self.TIME_LIMS = [kwargs["start_time"], kwargs["end_time"]]
+        kwargs["start_time"] -= self.DEFAULTS["pad_times"][0]
+        kwargs["end_time"] += self.DEFAULTS["pad_times"][-1]
+
+        if "EXPT" in self.COLLECTION:
+
+            from viresclient import SwarmRequest
+
+            SERVER_URL = "https://vires.services/ows"
+            request = SwarmRequest(SERVER_URL)
+            request.set_collection(self.COLLECTION)
+
+            if self.VARNAME == "Eh_XYZ":
+                var_list = ["Ehx", "Ehy", "Ehz", "Quality_flags"]
```

Comment:
I think it will be less confusing to avoid using special variables like `Eh_XYZ` used here that doesn't exist in the input data